CMakeLists.txt

```diff
@@ -0,0 +1,108 @@
+# Project information.
+cmake_minimum_required( VERSION 3.5.0 )
+project( AwsIotDeviceSdkEmbeddedC
+         VERSION 4.0.2
+         LANGUAGES C )
+
+# Allow the project to be organized into folders.
+set_property( GLOBAL PROPERTY USE_FOLDERS ON )
+
+# Use C90.
+set( CMAKE_C_STANDARD 90 )
+set( CMAKE_C_STANDARD_REQUIRED ON )
+
+# Do not allow in-source build.
+if( ${PROJECT_SOURCE_DIR} STREQUAL ${PROJECT_BINARY_DIR} )
+    message( FATAL_ERROR "In-source build is not allowed. Please build in a separate directory, such as ${PROJECT_SOURCE_DIR}/build." )
+endif()
+
+# Import global configurations.
+include("tools/cmake/filePaths.cmake")
+
+# Configure options to always show in CMake GUI.
+option( BUILD_TESTS
+        "Set this to ON to build both demo and test executables. When OFF, only demo executables are built."
+        ON )
+option( BUILD_CLONE_SUBMODULES
```

Comment:
I think this option can also be avoided. (Imo) We should start with the minimal feature set of build options for CMake, and add more if needed in the future.

Reply:
We should think about this, my best guess this was added becuase building including the cloning of submodules requires a round-trip TLS connection to github for every submodule which is something which can easily add minutes or even hours to the nightly CI. 

For Context the ESP-IDF submodule is 180MB, the Microchip Harmony one is also huge, I was looking at the Cypress one which is 150MB last week ...

Is there a different way to give the option to not re-fetch the submodules every time?

Reply:
I am having cmake update submodules from now on whenever this option is activated.